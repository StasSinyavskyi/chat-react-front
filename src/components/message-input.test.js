/* eslint-env jest */
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import MessageInput from './message-input';

// if we have several test we can join them in describe
describe('<MessageInput />', () => {
  const testParam = (
    <MessageInput
      onJoinButtonClick={() => 'onJoinButtonClick'}
      sendMessage={() => 'sendMessage'}
      showJoinButton={false}
      disabled={false}
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const tree = renderer
      .create(testParam)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // test2
  const testParam2 = (
    <MessageInput
      onJoinButtonClick={() => 'onJoinButtonClick'}
      sendMessage={() => 'sendMessage'}
      showJoinButton
      disabled={false}
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam2, div);
    ReactDOM.unmountComponentAtNode(div);
  });


  it('render correctly', () => {
    const tree = renderer
      .create(testParam2)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
