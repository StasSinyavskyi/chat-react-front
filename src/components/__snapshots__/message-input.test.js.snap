// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> render correctly 1`] = `
<div
  className="MessageInput-messageInputWrapper-45"
>
  <div
    className="MuiPaper-root-47 MuiPaper-shadow6-55 MuiPaper-rounded-48 MessageInput-messageInput-46"
  >
    <div
      className="MuiInput-root-74 MuiInput-fullWidth-82 MuiInput-inkbar-76 MuiInput-underline-80"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
    >
      <input
        aria-invalid={undefined}
        aria-required={undefined}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-83"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name={undefined}
        onChange={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your messageâ€¦"
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput /> render correctly 2`] = `
<div
  className="MessageInput-messageInputWrapper-45"
>
  <div
    className="MuiPaper-root-47 MuiPaper-shadow6-55 MuiPaper-rounded-48 MessageInput-messageInput-46"
  >
    <button
      className="MuiButtonBase-root-128 MuiButton-root-113 MuiButton-raised-118 MuiButton-raisedPrimary-120 MuiButton-fullWidth-127"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label-114"
      >
        Join
      </span>
      <span
        className="MuiTouchRipple-root-130"
      />
    </button>
  </div>
</div>
`;
